x-pack/plugins/fleet/server/services/epm/packages/custom_integrations/validation/check_naming_collision.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { nodeBuilder } from '@kbn/es-query';
+import type { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
+
+import { i18n } from '@kbn/i18n';
+
+import { PackageNotFoundError } from '../../../../../errors';
+
+import { auditLoggingService } from '../../../../audit_logging';
+import { PACKAGES_SAVED_OBJECT_TYPE, type Installation } from '../../../../../../common';
+import * as Registry from '../../../registry';
+
+export const checkForNamingCollision = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  integrationName: string
+) => {
+  try {
+    await Registry.fetchFindLatestPackageOrThrow(integrationName);
```

Comment:
Wouldn't it be simpler to use `fetchFindLatestPackageOrUndefined` and check if the result is undefined? It's not that easy to understand first why we are not using the result of `fetchFindLatestPackageOrThrow` as the logic continues in line 40-41.

Reply:
Sure, I can switch over to `undefined` vs `throw`.